apps/web/src/components/workflow/preview/email/EmailPreview.tsx

```diff
@@ -1,207 +1,125 @@
-import { Grid, JsonInput, useMantineTheme } from '@mantine/core';
-import { useCallback, useEffect, useMemo, useState } from 'react';
-import { useFormContext, useWatch } from 'react-hook-form';
-
+import { JsonInput } from '@mantine/core';
+import { useEffect, useMemo, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import styled from '@emotion/styled';
 import { Button, colors, inputStyles } from '@novu/design-system';
 
-import { IS_DOCKER_HOSTED, useDataRef } from '@novu/shared-web';
-import { useGetLocalesFromContent, usePreviewEmail } from '../../../../api/hooks';
-import { useActiveIntegrations, useAuthController, useProcessVariables } from '../../../../hooks';
+import { useActiveIntegrations, useProcessVariables } from '../../../../hooks';
 import type { IForm } from '../../../../pages/templates/components/formTypes';
 import { useStepFormErrors } from '../../../../pages/templates/hooks/useStepFormErrors';
 import { useStepFormPath } from '../../../../pages/templates/hooks/useStepFormPath';
 import { When } from '../../../utils/When';
 import { PreviewMobile } from './PreviewMobile';
 import { PreviewWeb } from './PreviewWeb';
+import { useTemplateLocales } from '../../../../pages/templates/hooks/useTemplateLocales';
+import { usePreviewEmailTemplate } from '../../../../pages/templates/hooks/usePreviewEmailTemplate';
+
+const PreviewContainer = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  height: 100%;
+  margin: 0;
+  margin-bottom: 1.5rem;
+  gap: 1.5rem;
+`;
+
+const JSONContainer = styled.div`
+  min-width: 28rem;
+  height: 100%;
+  background: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.B17 : colors.B98)};
+  border-radius: 7;
+`;
```

Comment:
I also did a little refactoring here. Use custom styled components instead of the Mantine grid components as it had many unnecessary styles that were breaking the layout.